packages/framework/src/utils/platform.utils.ts

```diff
@@ -0,0 +1,8 @@
+import type { IResponseError } from '@novu/shared';
+
+/**
+ * Validate (type-guard) that an error response matches our IResponseError interface.
+ */
+export const checkIsResponseError = (err: unknown): err is IResponseError => {
+  return !!err && typeof err === 'object' && 'error' in err && 'message' in err && 'statusCode' in err;
+};
```

Comment:
@SokratisVidros I have duplicated this from `@novu/shared` to resolve the error reported [here](https://discord.com/channels/895029566685462578/1312279985192435732/1312279985192435732) in Discord and elsewhere when using Framework in Remix & NextJS. Enums may be an issue but this could resolve it.